datasource-local/src/main/java/cmm/apps/esmorga/datasource_local/event/EventLocalDatasourceImpl.kt

```diff
@@ -26,4 +26,26 @@ class EventLocalDatasourceImpl(private val eventDao: EventDao) : EventDatasource
     override suspend fun deleteCacheEvents() {
         eventDao.deleteAll()
     }
+
+    override suspend fun joinEvent(eventId: String) {
+        val events = eventDao.getEvents().map {
```

Comment:
This is not optimized, better to create a getEventById and avoid getting the whole list and go through it.